src/main/java/mate/academy/dao/BookDaoimpl.java

```diff
@@ -0,0 +1,121 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement ps = connection.prepareStatement(sql)) {
+            ps.setLong(1, id);
+            ResultSet rs = ps.executeQuery();
+            if (rs.next()) {
+                Book book = getBookFromResultSet(rs);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create a connection to the DB", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String sql = "SELECT * FROM book";
+        List<Book> books = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement ps = connection.prepareStatement(sql)) {
+            ResultSet rs = ps.executeQuery();
+            while (rs.next()) {
+                Book book = getBookFromResultSet(rs);
+                books.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot find any books", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE book SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement ps = connection.prepareStatement(sql)) {
+            ps.setString(1, book.getTitle());
+            ps.setBigDecimal(2, book.getPrice());
+            ps.setLong(3, book.getId());
+            int affectedRows = ps.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException("Failed to update the book with id: " + book.getId());
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot update book", e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String sql = "DELETE FROM book WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement ps = connection.prepareStatement(sql)) {
+            ps.setLong(1, id);
+            int affectedRows = ps.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException("Failed to delete the book with id: " + id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot delete book", e);
+        }
+        return true;
+    }
+
+    private Book getBookFromResultSet(ResultSet resultSet) throws SQLException {
+        Book book = new Book();
+        Long id = resultSet.getObject("id", Long.class);
+        String title = resultSet.getString("title");
+        BigDecimal price = resultSet.getBigDecimal("price");
+        book.setId(id);
+        book.setTitle(title);
+        book.setPrice(price);
+        return book;
```

Comment:
Consider refactoring `getBookFromResultSet` to use a constructor or setters consistently, but not both. For example, use `new Book(id, title, price)` or use setters for all fields.